scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -614,7 +618,8 @@ class ScioContext private[scio] (val options: PipelineOptions,
    */
   def pubsubTopic[T: ClassTag](topic: String,
                                idLabel: String = null,
-                               timestampLabel: String = null)
+                               timestampLabel: String = null,
+                               parseFn: gio.PubsubIO.PubsubMessage => T = null)
```

Comment:
I was going to say I see a use for both, as I would imagine use cases where the pubsub attributes describe the format of the payload, but in that case `T` can just be `Array<Byte>` and the decoding can happen in a separate step.